### PR TITLE
Improve inflate parameters

### PR DESCRIPTION
**Pull request by clallemand**
#### New features

- Add options in `inflate_parameters` and `inflate_parameter_leaf`:
  - `start_update_instant` : Instant of the year when the inflation should start, if different from January 1st
  - `round_ndigits` : number of digits in the rounded result
- Adjustment of `inflate_parameters` to use it with parameter leaf.

### CODE REVIEW

File: openfisca_survey_manager/utils.py [hunk 1]
```diff
@@ -12,59 +12,59 @@
...
 
 
-def inflate_parameters(parameters, inflator, base_year, last_year = None, ignore_missing_units = False):
+def inflate_parameters(parameters, inflator, base_year, last_year = None, ignore_missing_units = False,
+                       start_update_instant = None, round_ndigits = 2):
 
     if (last_year is not None) and (last_year > base_year + 1):
         for year in range(base_year + 1, last_year + 1):
-            inflate_parameters(parameters, inflator, year - 1, last_year = year, ignore_missing_units = ignore_missing_units)
+            inflate_parameters(parameters, inflator, year - 1, last_year = year, ignore_missing_units = ignore_missing_units,
```
Comment (benoit-cty) [inline openfisca_survey_manager/utils.py:20]: Is it possible to add an explanation that there is a recursive call and how it works and what is the out condition ?
I understand that we inflate n years with the inflation of first year, then n-1 with inflation of second year and so on until we reach the last year, right ?

File: openfisca_survey_manager/utils.py [hunk 2]
```diff
@@ -12,59 +12,59 @@
...
-
-                for unit_type in unit_by_type.keys():
-                    if sub_parameter.metadata[unit_type].startswith("currency"):
-                        inflate_parameter_leaf(sub_parameter, base_year, inflator, unit_type = unit_type)
-
-
-def inflate_parameter_leaf(sub_parameter, base_year, inflator, unit_type = 'unit'):
+        if isinstance(parameters, ParameterNode):
+            for sub_parameter in parameters.children.values():
+                inflate_parameters(sub_parameter, inflator, base_year, last_year, ignore_missing_units = ignore_missing_units,
```
Comment (benoit-cty) [inline openfisca_survey_manager/utils.py:31]: Is it a second recursive call ?

Comment (clallemand) [reply openfisca_survey_manager/utils.py:31]: Yes, to inflate each parameter leaf of the given parameter node

File: openfisca_survey_manager/utils.py [hunk 3]
```diff
@@ -12,59 +12,59 @@
 log = logging.getLogger(__name__)
 
 
-def inflate_parameters(parameters, inflator, base_year, last_year = None, ignore_missing_units = False):
+def inflate_parameters(parameters, inflator, base_year, last_year = None, ignore_missing_units = False,
+                       start_update_instant = None, round_ndigits = 2):
```
Comment (benjello) [inline openfisca_survey_manager/utils.py:16]: Use start_instant instead of `start_update_instant`

File: openfisca_survey_manager/utils.py (hunk 1 above)
Comment (benjello) [reply openfisca_survey_manager/utils.py:20]: Yep

**Review by benjello — CHANGES_REQUESTED** [review]
@clallemand looks good !
Please consider adding tests of your options in this [file](https://github.com/openfisca/openfisca-survey-manager/blob/master/openfisca_survey_manager/tests/test_legislation_inflator.py)

File: openfisca_survey_manager/tests/test_legislation_inflator.py [hunk 4]
```diff
@@ -84,9 +84,31 @@ def test_inflate_scale_with_changing_number_of_brackets():
             )
 
 
+def test_inflate_start_instant_option():
+    """
+        Test parameters inflator with a specific start_instant
+    """
+    tax_benefit_system = CountryTaxBenefitSystem()
+    parameters = tax_benefit_system.parameters
+    parameters_asof(parameters, instant = periods.instant(2022))  # Remove post 2016 legislation changes
```
Comment (benjello) [inline openfisca_survey_manager/tests/test_legislation_inflator.py:93]:
```suggestion
    parameters_asof(parameters, instant = periods.instant(2022))  # Remove post 2022 legislation changes
```

File: setup.py [hunk 5]
```diff
@@ -30,7 +30,7 @@
 
 setup(
     name = 'OpenFisca-Survey-Manager',
-    version = '1.0.2',
+    version = '1.0.3',
```
Comment (benjello) [inline setup.py:33]:
```suggestion
    version = '1.1.0',
```

**Review by benjello — APPROVED** [review]
Please accept my suggestions then merge.

**Pull request merged**